pkg/git/git.go

```diff
@@ -244,3 +253,26 @@ func userHasKnownHostsFile(logger *zap.SugaredLogger) (bool, error) {
 	f.Close()
 	return true, nil
 }
+
+func validateGitAuth(logger *zap.SugaredLogger, url string) error {
+	homeenv := os.Getenv("HOME")
+	sshCred := true
+	if _, err := os.Stat(homeenv + "/.ssh"); os.IsNotExist(err) {
+		sshCred = false
+	}
+	urlSSHFormat := validateGitSSHURLFormat(url)
+	if sshCred && !urlSSHFormat {
+		logger.Warnf("The given URL does not match the ssh credential")
+	} else if !sshCred && urlSSHFormat {
+		logger.Warnf("The given ssh URL does not match the credential")
+	}
+
+	return nil
```

Comment:
We don't seem to ever return an error from this func, only print warnings.  Not sure if `error` is needed in function signature.